ToolBox/FactorFileGenerator.cs

```diff
@@ -226,11 +226,16 @@ private FactorFileRow CalculateNextDividendFactor(BaseData dividend, FactorFileR
 
             TradeBar previousClosingPrice = FindPreviousTradableDayClosingPrice(eventDayData.Time);
 
-            var priceFactor = previousFactorFileRow.PriceFactor - (dividend.Value / ((previousClosingPrice.Close) * previousFactorFileRow.SplitFactor));
+            // round split factor back to natural number
+            var splitFactorMultiplier = Math.Round(1 / previousFactorFileRow.SplitFactor);
```

Comment:
this has been reverted. I thought that splits sould only be happening in the direction of multiple shares increase? how would otherwise to convert the prime number of shares when uncommon splits occur? Suppose one holds 7 shares how they are being converted on 2 to 3 split? but I am sure that there must be a mechanism to do it..

Reply:
Great question. In the exotic cases, partial-share shareholders are typically paid out in cash at the previous day's closing market rate.